### PR TITLE
Sector boundary modifications and sector file cleanup

### LINKED ISSUE #45: Sector Cleanup
- Current .sct and .ese is full of badly drawn and/or duplicated and/or inaccurate and/or awkwardly named lines
- Current internal sectorisation is not very good (the new one is closer, but we don't have the coordinates so it probably isn't completely accurate)

### PR DESCRIPTION
**Pull request by arthuwu**
Resolves #45 

### Sector boundaries
- All boundaries have been redrawn and made more accurate, referencing HK AIP ENR6 and various images of the real life ATM. These changes do not impact transfer of control points/COPX defined in LOAs/SOPs.
- Notable changes:
    - DEH/TRW north-west (TAMOT) boundary with ZGZU
    - TMW/TRW northern (SIERA/ROMEO/LANDA) boundary with ZGZU
    - TRE/TME northern (north of J101) boundary
       - The DOTMI/LELIM section does not correspond closely to real life due to our LOA
    - APP/DEP northern (BEKOL) boundary with ZGZU
    - 07 and 25 FAD boundaries
    - 07 APP/DEP noise abatement sectorisation
 
### NAPs
- Sectorisation and SID maps for noise abatement procedures (runway 07 operations) at VHHH have been implemented. Controllers can toggle between normal and noise abatement sectorisation by selecting the "DAY" or "NAP" pseudo-runways in Euroscope. 

### CTR and UCARA
- The Hong Kong CTR and UCARA and their subdivisions have been redrawn to match AIP coordinates (#52)

### MRVA
- 9000' (D50 BIGEX Class C) ring added
- 3000' (D25 MCU + D30 VHHH ARP) rings corrected 
> - (2b72535) (#47) More accurate display format (already published in 2313v1)

### Display/Maps
- DEH subsectors are no longer drawn by default and are now togglable TopSky dashed lines
- The entire northern boundary of the appropriate APP/DEP sector now displays when any controller covering adjacent PRC sector(s) are online - previously only a small section was displayed
- 2nm buffer between MCU/APP/DEP and ZGZU is now drawn on the TMA ASR, as per the real life ATM - although it is at a different distance due to our LOA
- CTR, APP and ZNC ASRs converted to magnetic North up, as per the real life ATM
> - (#49) Danger areas are now drawn with TopSky and will activate based on time of day/NOTAM referencing the HK AIP

### Cleanup
- Unused color definitions have been removed
- Several duplicated lines have been removed
> - (#50) Coastlines updated (24833 less lines) (already published in 2313v1)

### CODE REVIEW

**Comment by arthuwu** [comment]
## To do
- [ ] Modify TopSkyMaps.txt to use TopSky dotted lines for DEH subsectors
- [ ] Clean up .sct colour definitions
- [ ] Update .ese to match new sector boundaries
+
- [ ] Custom font? maybe

File: Data/Sector/Hong-Kong-Sector-File.sct [hunk 1]
```diff
@@ -6254,5748 +6138,368 @@ TRAN/TAMOT->SIKOU         N000.00.00.000 E000.00.00.000 N000.00.00.000 E000.00.0
...
-                                         N022.15.45.323 E114.15.21.239 N022.15.50.111 E114.15.33.120 TWR&CTR
-                                         N022.15.54.900 E114.15.45.000 N022.15.59.688 E114.15.56.879 TWR&CTR
-                                         N022.16.04.476 E114.16.08.759 N022.16.09.263 E114.16.20.640 TWR&CTR
-                                         N022.16.14.051 E114.16.32.520 N022.16.18.839 E114.16.44.399 TWR&CTR
-                                         N022.16.23.628 E114.16.56.280 N022.16.28.416 E114.17.08.160 TWR&CTR
-                                         N022.16.33.204 E114.17.20.039 N022.16.37.991 E114.17.31.919 TWR&CTR
-                                         N022.16.42.779 E114.17.43.800 N022.16.47.567 E114.17.55.679 TWR&CTR
-                                         N022.16.52.356 E114.18.07.559 N022.16.57.144 E114.18.19.440 TWR&CTR
-                                         N022.17.01.932 E114.18.31.320 N022.17.06.719 E114.18.43.199 TWR&CTR
-
```
Comment (EightSmart) [inline Data/Sector/Hong-Kong-Sector-File.sct:6241]: I see you've drawn these intentionally to avoid having any lines overlap, but results in it being difficult to toggle certain boundaries on and off if controllers desire to. For example if for whatever reason I want to create a TMA asr just for controlling the TME sector, I can't, because it looks like this:
image

I'd recommend either adding all of the extra lines back in so each sector is "complete" when displayed. If the overlapping lines looks too messy (which they shouldn't if they are exactly the same?) then we could make a "ALL BOUNDARIES" GEO that contains all of this version, which is displayed by default. Then have each of the individual sectors drawn in full separately available for toggle as necessary.

File: Hong Kong TOPSKY.prf [hunk 2]
```diff
@@ -62,12 +62,12 @@ Settings	sector	\Data\Sector\Hong-Kong-Sector-File.sct
...
-RecentFiles	Recent3	\Data\ASR\Hong Kong TMA.asr
-RecentFiles	Recent4	\Data\ASR\Kai Tak GMC.asr
-RecentFiles	Recent5	\Data\ASR\Macau GMC.asr
-RecentFiles	Recent6	\Data\ASR\Hong Kong Zone.asr
+RecentFiles	Recent1	\Data\ASR\Kai Tak GMC.asr
+RecentFiles	Recent2	\Data\ASR\Hong Kong ACC.asr
+RecentFiles	Recent3	\Data\ASR\Hong Kong Zone.asr
+RecentFiles	Recent4	\Data\ASR\Macau GMC.asr
+RecentFiles	Recent5	\Data\ASR\Hong Kong GMC.asr
+RecentFiles	Recent6	\Data\ASR\Hong Kong TMA.asr
```
Comment (EightSmart) [inline Hong Kong TOPSKY.prf:70]: Revert the changes to this file to avoid unnecessary conflicts when people update with the installer

File: Data/Sector/Hong-Kong-Sector-File.ese [hunk 3]
```diff
@@ -617,18 +621,18 @@ DISPLAY:VHHK·VHHK_SCT_25_DEH_02·100·250:VHHK·VHHK_SCT_25_DEH_02·100·250:VH
 DISPLAY:VHHK·VHHK_SCT_25_DEH_02·100·250:VHHK·VHHK_SCT_25_DEH_02·100·250:VHHK·VHHK_SCT_25_DEH_01·135·250
 DISPLAY:VHHK·VHHK_SCT_25_DEH_02·100·250:VHHK·VHHK_SCT_25_DEH_02·100·250:VHHK·VHHK_SCT_25_DEP_03·000·135
 DISPLAY:VHHK·VHHK_SCT_25_DEP_02·000·100:VHHK·VHHK_SCT_25_DEP_02·000·100:VHHK·VHHK_SCT_07_FAD_01·000·100
-DISPLAY:VHHK·VHHK_SCT_25_DEP_02·000·100:VHHK·VHHK_SCT_25_DEP_02·000·100:VHHK·VHHK_SCT_07_DEP_01·000·135
+DISPLAY:VHHK·VHHK_SCT_25_DEP_02·000·100:VHHK·VHHK_SCT_25_DEP_02y·000·100:VHHK·VHHK_SCT_07_DEP_01·000·135
```
Comment (EightSmart) [inline Data/Sector/Hong-Kong-Sector-File.ese:624]: typo

File: Data/Sector/Hong-Kong-Sector-File.ese [hunk 4]
```diff
@@ -2453,14 +1982,179 @@ COORD:N023.30.00.000:E117.30.00.000
...
+COORD:N022.34.50.000:E114.16.05.000
+COORD:N022.34.47.978:E114.18.13.952
+COORD:N022.36.32.635:E114.23.53.000
+COORD:N022.32.27.709:E114.28.00.598
+COORD:N022.27.43.342:E114.29.03.948
+COORD:N022.26.56.963:E114.30.08.265
+COORD:N022.23.50.616:E114.30.47.871
+COORD:N022.28.17.000:E114.41.57.470
+DISPLAY:VHHK·VHHK_SCT_25_FAD_01·000·085:VHHK·VHHK_SCT_25_FAD_01·000·085:ZGZU·ZGGG·000·256
+DISPLAY:VHHK·VHHK_SCT_25_DEH_01·135·250:VHHK·VHHK_SCT_25_DEH_01·135·250:ZGZU·ZGGG·000·256
```
Comment (EightSmart) [inline Data/Sector/Hong-Kong-Sector-File.ese:2050]: Extra line here
Config below: on VHHH_F_APP, other controllers online shown
image

Also, shouldn't this also display if Zhuhai APP is online? (BEKOL transfer)

File: Data/Sector/Hong-Kong-Sector-File.ese [hunk 5]
```diff
@@ -2779,6 +2432,96 @@ ACTIVE:VMMC:16
...
+ARRAPT:VHHH
+
+SECTOR:VHHK·VHHK_SCT_07NAP_DEH_05·165·250:13500:25000
+OWNER:DEH:DEP:APP:FAD:TRE:PAR
+ALTOWNER:Mentoring VHHH_APP:DEH:DEP:INS:MTR
+ALTOWNER:Mentoring HKG_E_CTR:DEH:DEP:APP:FAD:INS:MTR
+BORDER:112:111:21:22:23:24
+ACTIVE:VHHH:NAP
+DEPAPT:VHHH
+ARRAPT:VHHH
```
Comment (EightSmart) [inline Data/Sector/Hong-Kong-Sector-File.ese:2525]: ### (MAJOR PRIORITY)
We need to write new COPX rules for ALL the new noise abatement sectors

File: Data/Sector/Hong-Kong-Sector-File.ese [hunk 6]
```diff
@@ -1174,8 +1192,10 @@ DISPLAY:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VH
 DISPLAY:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VHHK_SCT_25_DEP_05·135·250
 DISPLAY:VHHK·VHHK_SCT_TRW_01·250·365:VHHK·VHHK_SCT_TRW_01·250·365:VHHK·VHHK_SCT_TRE_01·250·365
 DISPLAY:VHHK·VHHK_SCT_TRE_01·250·365:VHHK·VHHK_SCT_TRW_01·250·365:VHHK·VHHK_SCT_TRE_01·250·365
+DISPLAY:VHHK·VHHK_SCT_07NAP_DEP_03·000·250:VHHK·VHHK_SCT_07NAP_DEP_03·000·250:VHHK·VHHK_SCT_TME_01·000·250
+DISPLAY:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VHHK_SCT_07NAP_DEP_03·000·250:VHHK·VHHK_SCT_07NAP_DEP_03·000·250
```
Comment (EightSmart) [inline Data/Sector/Hong-Kong-Sector-File.ese:1196]: Sectorline 32 and 35 (below) not displayed/missing in NAP configuration (syntax error)?

Example: Logged on as HKG_E_CTR, own TME, NAP 07 Configuration, VHHH_APP online
image
```suggestion
DISPLAY:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VHHK_SCT_07NAP_DEP_03·000·250
```

File: Data/Sector/Hong-Kong-Sector-File.ese [hunk 7]
```diff
@@ -1229,8 +1249,10 @@ DISPLAY:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VH
 DISPLAY:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VHHK_SCT_25_DEP_06·135·250
 DISPLAY:VHHK·VHHK_SCT_TRE_01·250·365:VHHK·VHHK_SCT_TRE_01·250·365:VHHK·VHHK_SCT_TRE_02·250·365
 DISPLAY:VHHK·VHHK_SCT_TRE_02·250·365:VHHK·VHHK_SCT_TRE_02·250·365:VHHK·VHHK_SCT_TRE_01·250·365
+DISPLAY:VHHK·VHHK_SCT_07NAP_DEP_03·000·250:VHHK·VHHK_SCT_07NAP_DEP_03·000·250:VHHK·VHHK_SCT_TME_01·000·250
+DISPLAY:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VHHK_SCT_07NAP_DEP_03·000·250:VHHK·VHHK_SCT_07NAP_DEP_03·000·250
```
Comment (EightSmart) [inline Data/Sector/Hong-Kong-Sector-File.ese:1253]: See comment above about SECTORLINE 32
```suggestion
DISPLAY:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VHHK_SCT_TME_01·000·250:VHHK·VHHK_SCT_07NAP_DEP_03·000·250
```

**Review by EightSmart — CHANGES_REQUESTED** [review]

File: Data/ASR/Hong Kong ACC.asr [hunk 8]
```diff
@@ -3,22 +3,23 @@ DisplayTypeNeedRadarContent:1
...
-ARTCC boundary:ZGNN - Nanning ACC:
-ARTCC boundary:ZJSY - Sanya ACC:
-ARTCC boundary:ZSAM - Xiamen ACC:
-ARTCC boundary:ZSSS - Shanghai ACC:
-ARTCC low boundary:5865472 - 5865580 8421493 - VHHH | Hong Kong:
-ARTCC low boundary:Terminal Control 8421493 - VMMC | Macau:
-ARTCC low boundary:VHHH TME_(EAST_/_SFC-FL250):
-ARTCC low boundary:VHHH TMS_(SOUTH_/_SFC-FL250):
-ARTCC low boundary:VHHH TMW_(WEST_/_FL120-FL250):
-Geo:VHHH MRVA Min Vector Alts:
```
[the remark this answers is not in the document]
Comment (arthuwu) [reply Data/ASR/Hong Kong ACC.asr:21]: Will be fixed in updated colour scheme/ASRs - according to real life references these and the MRVA lines are not shown on area positions

File: Data/Plugins/TopSky/TopSkyMaps.txt [hunk 9]
```diff
@@ -422,7 +441,7 @@ ASRDATA:IFR,ILS
 FOLDER:VHHH
 COLOR:STAR
 ACTIVE:RWY:ARR:VHHH25R:DEP:*
-STYLE:DOT:1
+STYLE:SOLID:1
```
[the remark this answers is not in the document]
Comment (arthuwu) [reply Data/Plugins/TopSky/TopSkyMaps.txt:444]: The missed path doesn't appear to be drawn in any of my reference images, I'd be happy to curve it, but I was actually planning on excluding it from the ASR by default - any thoughts on this?

File: Data/ASR/Hong Kong ACC.asr [hunk 10]
```diff
@@ -147,10 +148,10 @@ HISTORY_DOTS:0
...
 DISABLEZOOMING:0
-DisplayRotation:0.00000
+DisplayRotation:2.00000
 TAGFAMILY:Hong Kong Radar TOPSKY
 WINDOWAREA:17.343885:108.683889:23.319267:120.443485
 PLUGIN:CoFrance:RAD:1
 PLUGIN:CoFrance:VV:1
 PLUGIN:CoFrance:VVTIME:2
-PLUGIN:TopSky plugin:HideMapData:CSTR,SID,STAR,WS,VFR
+PLUGIN:TopSky plugin:HideMapData:CSTR,SID,STAR,WS,VFR,BUFFER
```
[the remark this answers is not in the document]
Comment (arthuwu) [reply Data/ASR/Hong Kong ACC.asr:157]: All SIDs and STARs and no extended centerlines appear in my area references - will implement this in the colour scheme update

image
image

File: Hong Kong TOPSKY.prf (hunk 2 above)
Comment (arthuwu) [reply Hong Kong TOPSKY.prf:70]: Sorry - didn't realise I committed this

File: Data/Sector/Hong-Kong-Sector-File.sct (hunk 1 above)
Comment (arthuwu) [reply Data/Sector/Hong-Kong-Sector-File.sct:6241]: Will add the missing lines in

File: Data/Sector/Hong-Kong-Sector-File.ese (hunk 5 above)
Comment (arthuwu) [reply Data/Sector/Hong-Kong-Sector-File.ese:2526]: New COPX:
 07 NAP DEP -> 07 NAP APP 
 9000 GUAVA

File: Data/Sector/Hong-Kong-Sector-File.sct (hunk 1 above)
[the remark this answers is not in the document]
Comment (arthuwu) [reply Data/Sector/Hong-Kong-Sector-File.sct:6298]: Fixed - moved actual boundaries to ARTCC and FIR boundary to ARTCC High because Euroscope is strange

File: Data/Plugins/TopSky/TopSkyMaps.txt (hunk 9 above)
[the remark this answers is not in the document]
Comment (EightSmart) [reply Data/Plugins/TopSky/TopSkyMaps.txt:444]: Let's add it in for now, but we can disable displaying the missed approach by default

**Review by EightSmart — APPROVED** [review]

**Review by EightSmart — APPROVED** [review]
